storage: minor cleanups around pebble compaction concurrency

RaduBerinde opened the pull request:
#### storage: remove AdjustCompactionConcurrency

This is not used.

Epic: none
Release note: None

#### storage: remove unnecessary initialization in newPebble

We now only call this function from Open and we always have options
set.

Epic: none
Release note: None

#### pebble: separate compaction concurrency override logic

Separate the logic that allows temporarily overriding compaction
concurrency.

Epic: none
Release note: None

#### storage: cleanup default max concurrent compactions code

Refactor this code so we statically initialize the default value.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/145288)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145288#-:-OP1oYj66z59bRB8IKPA:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @annrpom and @jbowens)

___
*[`pkg/kv/kvserver/stores_server.go` line 202 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/145288#-OP1_OPX6nWC27hd8DAh:-OP1_OPX6nWC27hd8DAi:b-frd63y) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b65729b531b6f90bba7a488348002d0c2ec245e/pkg/kv/kvserver/stores_server.go#L202)):*
> ```Go
> 	err := is.execStoreCommand(ctx, req.StoreRequestHeader,
> 		func(ctx context.Context, s *Store) error {
> 			s.TODOEngine().SetCompactionConcurrency(req.CompactionConcurrency)
> ```

what happens if there are two concurrent calls:
goroutine 1: setCompactionConcurrency(10)
goroutine 2: setCompactionConcurrency(5)
goroutine 1: (cancelled) set CompactionConcurrency(0)
...
goroutine 2: (cancelled) setCompactionConcurrency(0)

I don't think the behavior was clean before this change either. Probably worth documenting that we don't stack the calls up, and that the behavior is not defined except for the fact that the original value will be restored.
___
*[`pkg/storage/engine.go` line 1107 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/145288#-OP1nSxl3d9afEoRQCvA:-OP1nSxl3d9afEoRQCvB:b-iss1yf) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b65729b531b6f90bba7a488348002d0c2ec245e/pkg/storage/engine.go#L1107)):*
> ```Go
> 	// SetCompactionConcurrency is used to override the engine's max compaction
> 	// concurrency. A value of 0 removes any existing override.
> 	SetCompactionConcurrency(n uint64)
> ```

could you add a simple unit test of this functionality.


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
pebble: separate compaction concurrency override logic

Separate the logic that allows temporarily overriding compaction
concurrency.

Epic: none
Release note: None

RaduBerinde committed:
storage: remove AdjustCompactionConcurrency

This is not used.

Epic: none
Release note: None

RaduBerinde committed:
storage: remove unnecessary initialization in newPebble

We now only call this function from Open and we always have options
set.

Epic: none
Release note: None

RaduBerinde committed:
storage: cleanup default max concurrent compactions code

Refactor this code so we statically initialize the default value.

Epic: none
Release note: None

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145288#-:-OP1sHQ37gMaIen90KNc:b-814g8k)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @annrpom, @jbowens, and @sumeerbhola)

___
*[`pkg/kv/kvserver/stores_server.go` line 202 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/145288#-OP1_OPX6nWC27hd8DAh:-OP1sahi6ysyVM4UkdoW:b3clt65) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b65729b531b6f90bba7a488348002d0c2ec245e/pkg/kv/kvserver/stores_server.go#L202)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

what happens if there are two concurrent calls:
goroutine 1: setCompactionConcurrency(10)
goroutine 2: setCompactionConcurrency(5)
goroutine 1: (cancelled) set CompactionConcurrency(0)
...
goroutine 2: (cancelled) setCompactionConcurrency(0)

I don't think the behavior was clean before this change either. Probably worth documenting that we don't stack the calls up, and that the behavior is not defined except for the fact that the original value will be restored.
</blockquote></details>

Updated the function comment. It's better now, because before we could have not restored the original value.
___
*[`pkg/storage/engine.go` line 1107 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/145288#-OP1nSxl3d9afEoRQCvA:-OP1rtNrBdIvy9vh_C0b:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b65729b531b6f90bba7a488348002d0c2ec245e/pkg/storage/engine.go#L1107)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

could you add a simple unit test of this functionality.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
